lib/travis/build/script/elm.rb

```diff
@@ -0,0 +1,183 @@
+module Travis
+  module Build
+    class Script
+      class Elm < NodeJs
+        # Default NodeJS version to install
+        DEFAULT_NODE_VERSION = '10.13.0'
+
+        DEFAULTS = {
+          elm: '0.19.0',
+          elm_test: '0.19.0',
+          elm_format: '0.19.0'
+        }
+
+        ELM_TEST_REQUIRED_NODE_VERSION = '6.0.0'
+
+        def export
+          super
+          sh.export 'TRAVIS_ELM_VERSION', elm_version, echo: false
+          sh.export 'TRAVIS_ELM_TEST_VERSION', elm_test_version, echo: false
+          sh.export 'TRAVIS_ELM_FORMAT_VERSION', elm_format_version, echo: false
+        end
+
+        def configure
+          super
+
+          config[:node_js] ||= DEFAULT_NODE_VERSION
+        end
+
+        def announce
+          super
+          sh.cmd 'elm --version', echo: true
+          sh.cmd 'elm-test --version', echo: true
+
+          # elm-format doesn't have --version,
+          # but the first line of `elm-format --help` prints the version
+          sh.cmd 'elm-format --help | head -n 1', echo: true
```

Comment:
I also suggest adding a blank line after this one, so that it is visually distinct from the next command (which is typically user-defined `before_install`, or `install`).

Reply:
Added in 17d4a121f4d18126b59a902eac364b8711395d63